src/NSubstitute/Extensions/ReceivedExtensions.cs

```diff
@@ -2,6 +2,10 @@
 using System.Collections.Generic;
 using System.Linq;
 using NSubstitute.Core;
+using NSubstitute.Exceptions;
+
+// Disable nullability for client API, so it does not affect clients.
+#nullable disable annotations
```

Comment:
**question:** these don't need to be re-enabled at the end of the file?

Reply:
Nice catch! I presume you are asking because at the bottom we have a plenty of internal classes. Is that right?

If so, then I think it's safe to not to re-enable nullability annotations:

- We only disable annotations, nullability warnings are still enabled. It means that if our code is calling any other internal code and it violates nullability rules, it will be caught.
- The internal `Quantity` classes are not used anywhere else in the project, so it's fine to not annotate them.

If you see some other reasoning, please follow up!